Add a To Do List example with API round-trip

robinheinze opened the pull request:
This adds a full-circuit example complete with:

- Screen
  - Implements `FlatList`
  - injects a MST store
- Component
  - An example of a MobX `@observer`
  - Has two presets
- MST Store
  - Appears as a child of rootStore
  - Has a prop consisting of an array of other MST instances
  - Has several views, and a `flow` action
- MST Record
- API service
  - I went with TODOs over Github because it's a familiar example, and it was completely fake data. I used https://jsonplaceholder.typicode.com/. It supports the ability to do POSTs and PATCHs also if we want to do that in the future.

![to-do-list-example](https://user-images.githubusercontent.com/6894653/40443280-b15a8218-5e7a-11e8-84ed-a2b0aa3cd2ca.gif)

kevinvangelder commented:
Do we want this to live in the boilerplate? Could it be a standalone example repo?

robinheinze commented:
I think it would be hard for it to be a fully functional real-world example without the surrounding context of the boilerplate. @jamonholmgren or @GantMan, any thoughts?

jamonholmgren commented:
@kevinvangelder has a good question, but I think I agree with @robinheinze that it works best in the context of the boilerplate. 👍

robinheinze commented:
We need some way to show them how the new stack works though. Maybe we just use the Chain React App as a living example?

GantMan commented:
I know it's not a popular opinion.  But I recommend a plugin called "Bells and Whistles" that would basically add screens/app stuff to demo the capabilities.

If that's a "no-go" and I believe it usually is, then an example app is the next best thing.  The only problem is an example app will age as the boilerplate grows.

robinheinze commented:
So it sounds like, I'll close this, and we just make a new `Examples` repo for now?

robinheinze commented:
What if it wasn't a repo, but a directory in the boilerplate repo?

skellock commented:
I think these are all good suggestions!  My concern is: it involves work 😆.

The chain react app will have api, state, maps, lists, themes, shared components, non-shared components, tests, and stories.  Plus it'll be shipping.

It's the perfect example.

kevinvangelder commented:
Chain React App 2018 is using AppSync, so it doesn't demo MST. I would be okay with an Examples directory so long as we're confident that it doesn't show up in the end-user's generated project.

robinheinze commented:
@GantMan @skellock @carlinisaacson @kevinvangelder @jamonholmgren 

I moved all the example content out into a new repo: https://github.com/infinitered/ignite-bowser-examples. Since the CR app isn't using our regular stack, this can be a place to put any working examples we don't want cluttering up the boilerplate itself.

This PR now just contains a few minor tweaks. I moved some styles onto the Text component and into typography instead of defining in the screens), and added a navigation helper method. 

Let me know your thoughts.

skellock commented:
Good stuff, thanks!

zakpgrove commented:
@kevinvangelder Excited to see the AppSync integration example! Will you be publishing the example before CR?

kevinvangelder commented:
@zakpgrove We ended up pivoting to use Apollo instead, we just made the repo public. https://github.com/infinitered/ChainReactApp2018

If you're looking for AppSync examples, @GantMan's React State Museum is an excellent resource: https://github.com/GantMan/ReactStateMuseum/
